platform/admin/core/admin-core-api/src/main/java/org/codice/ddf/ui/admin/api/impl/SystemPropertiesAdmin.java

```diff
@@ -192,37 +198,54 @@ public void writeSystemProperties(Map<String, String> updatedSystemProperties) {
             LOGGER.warn("Exception while writing to system.properties file.", e);
         }
 
+        writeOutUsersDotPropertiesFile(userPropertiesFile);
+        writeOutUsersDotAttributesFile(userAttributesFile);
+    }
+
+    /*
+     * Writes user property data to the relevant file after replacing the default hostname where
+     * necessary.
+     */
+    private void writeOutUsersDotPropertiesFile(File userPropertiesFile) {
         try {
-            Properties userDotProperties = new Properties(userPropertiesFile);
+            Properties usersDotProperties = new Properties(userPropertiesFile);
 
-            if (!userDotProperties.isEmpty()) {
-                String oldHostValue = userDotProperties.getProperty(oldHostName);
+            if (!usersDotProperties.isEmpty()) {
+                String oldHostValue = usersDotProperties.getProperty(oldHostName);
 
                 if (oldHostValue != null) {
-                    userDotProperties.remove(oldHostName);
-                    userDotProperties.setProperty(System.getProperty(SystemBaseUrl.HOST),
+                    usersDotProperties.remove(oldHostName);
+                    usersDotProperties.setProperty(System.getProperty(SystemBaseUrl.HOST),
                             oldHostValue);
 
-                    userDotProperties.save();
+                    usersDotProperties.save();
                 }
 
             }
-
         } catch (IOException e) {
             LOGGER.warn("Exception while writing to users.properties file.", e);
         }
+    }
 
+    /*
+     * Writes security and claims data for the system-level user and default admin.
+     */
+    private void writeOutUsersDotAttributesFile(File userAttributesFile) {
         Map<String, Object> json = null;
         try (InputStream stream = Files.newInputStream(Paths.get(userAttributesFile.toURI()))) {
             json = MAPPER.parser()
                     .parseMap(stream);
+
+            addGuestClaimsProfileAttributes(json);
```

Comment:
I'm curious if the placement of this logic (currently *before* the hostname check) affects if the system needs a restart assuming the hostname remains the default `localhost` in order to save developers time?